controllers/provisioners/eks/scaling/launchtemplate.go

```diff
@@ -237,6 +239,37 @@ func (lt *LaunchTemplate) Drifted(input *CreateConfigurationInput) bool {
 		drift = true
 	}
 
+	if len(latestVersion.LaunchTemplateData.LicenseSpecifications) != len(input.LicenseSpecifications) {
+		log.Info("detected drift", "reason", "Number of LicenseSpecifications has changed", "instancegroup", lt.OwnerName)
+		drift = true
+	}
+	for i, v := range latestVersion.LaunchTemplateData.LicenseSpecifications {
```

Comment:
This is a bit of lazy drift check. The AWS API doesn't guarantee ordering of the license specs so we could end up in a situation where the API reorders the list causing the instance-manager to reconcile the node group.